static/src/javascripts/projects/common/modules/ui/tabs.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+
+import fastdom from 'lib/fastdom-promise';
+
+const NAV_CLASSES = [
+    'tabs__tab--selected',
+    'tone-colour',
+    'tone-accent-border',
+];
+
+const getTabTarget = (tab: HTMLElement): Promise<?string> =>
+    fastdom.read(() => tab.getAttribute('href'));
+
+const hidePane = (tab: HTMLElement, pane: HTMLElement): Promise<void> => {
+    const tabList: HTMLElement = (tab.parentNode: any);
+
+    return fastdom.write(() => {
+        if (tabList) {
+            tabList.setAttribute('aria-selected', 'false');
+            NAV_CLASSES.forEach(className =>
+                tabList.classList.remove(className)
+            );
+        }
+
+        pane.classList.add('u-h');
+    });
+};
+
+const showPane = (tab: HTMLElement, pane: HTMLElement): Promise<void> => {
+    const tabList: HTMLElement = (tab.parentNode: any);
+
+    return fastdom.write(() => {
+        if (tabList) {
+            tabList.setAttribute('aria-selected', 'true');
+            NAV_CLASSES.forEach(className => tabList.classList.add(className));
+        }
+
+        pane.classList.remove('u-h', 'modern-hidden');
+        pane.focus();
+    });
+};
+
+const init = (): Promise<void> => {
+    const findTabs = (): Promise<Array<HTMLElement>> =>
+        fastdom.read(() => [...document.querySelectorAll('.tabs')]);
+
+    return findTabs().then(tabs => {
+        tabs.forEach(tab => {
+            const nav = tab.querySelector('.js-tabs');
+
+            if (!nav) {
+                return;
+            }
+
+            const { tabsInitialized } = nav.dataset || {};
```

Comment:
do you want the default for `tabsInitialized` to be an object literal here? I think this will result in `tabsInitialized` being `undefined` if nav.dataset doesn't contain the property `tabsInitialized`...

You could assign the default like so...

`const { tabsInitialized = {} } = nav.dataset;`